packages/lisk-transactions/src/transactions/3_vote_transaction.ts

```diff
@@ -0,0 +1,423 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { getAddressAndPublicKeyFromPassphrase } from '@liskhq/lisk-cryptography';
+import { VOTE_FEE } from '../constants';
+import { TransactionError, TransactionMultiError } from '../errors';
+import { Status, TransactionJSON } from '../transaction_types';
+import { prependMinusToPublicKeys, prependPlusToPublicKeys } from '../utils';
+import {
+	isAccountArray,
+	validatePublicKeys,
+	validator,
+} from '../utils/validation';
+import {
+	BaseTransaction,
+	createBaseTransaction,
+	CreateBaseTransactionInput,
+	ENTITY_ACCOUNT,
+	EntityMap,
+	RequiredState,
+	TransactionResponse,
+} from './base';
+
+const PREFIX_UPVOTE = '+';
+const PREFIX_UNVOTE = '-';
+const MAX_VOTE_PER_ACCOUNT = 101;
+const MIN_VOTE_PER_TX = 1;
+const MAX_VOTE_PER_TX = 33;
+
+export interface VoteAsset {
+	readonly votes: ReadonlyArray<string>;
+}
+
+export interface CreateVoteAssetInput {
+	readonly unvotes?: ReadonlyArray<string>;
+	readonly votes?: ReadonlyArray<string>;
+}
+
+export type CastVoteInput = CreateBaseTransactionInput & CreateVoteAssetInput;
+
+export const voteAssetTypeSchema = {
+	type: 'object',
+	required: ['votes'],
+	properties: {
+		votes: {
+			type: 'array',
+			items: {
+				format: 'signedPublicKey',
+			},
+		},
+	},
+};
+
+export const voteAssetFormatSchema = {
+	type: 'object',
+	required: ['votes'],
+	properties: {
```

Comment:
should it also check `amount` format (make sure it's zero)

Reply:
this part only covers asset, so I will check in validateSchema stage separately

Reply:
Yes that's true, but mu intention was in the future we only need to validate what is included in the asset =0